### DISCUSSION TITLE
Problem with Micropython on ESP32-C3 Super Mini

### DISCUSSION THREAD

**Discussion by bcux** [micropython/micropython#14047] [ESP32]
I have a problem with Micropython (1.22.2) on ESP32-C3 Super Mini
```
from machine import Pin
import time
p=Pin(8,Pin.OUT)

while True:
    p.on()
    time.sleep(1)
    p.off()
    time.sleep(1)

```
The program runs without any problems when I enter it via the console (USB connection with terminal program or via rshell and repl).
However, if I save it via rshell as main.py in the ESP32-C3, it does not run: LED does not flash.
The version of Micropython does not matter. On ESP8266 or ESP32-WROOM it runs as main.py without any problems.

**Comment by robert-hh** [comment]
Are you sure that this code resides as main.py on the board? What do you get when typing the below line at REPL:

`open("main.py").read()`

**Reply by bcux** [reply]
Yes, I'm shure
```
/home/bcux> ls -la /pyboard/
   139 Jan  1 2000  boot.py
   125 Mar  6 17:51 main.py

```
I can edit it with nano (which is my editor for rshell) and:

```
/home/xc/esp> repl
Entering REPL. Use Control-X to exit.
>
MicroPython v1.22.2 on 2024-02-22; ESP32C3 module with ESP32C3
Type "help()" for more information.
>>> 
>>> open('main.py').read()
'from machine import Pin\nimport time\np=Pin(8,Pin.OUT)\n\nwhile True:\n    p.on()\n    time.sleep(1)\n    p.off()\n    time.sleep(1)\n'
>>> 


```

**Reply by robert-hh** [reply]
That's strange. Similar, on a ESP32C3, if I enter REPL with rshell and then push Ctrl-D, main.py is executed. What is the content of your boot.py file?

**Comment by bcux** [comment]
If I enter repl and press Ctrl-D, main.py starts, but it does not start, when the ESP32-C3 is powered up. The behavior is very different to ESP8266 and ESP32-WROOM where main.py is started when powering up.

boot.py is not changed, it is the same as after flashing the ESP
```
# This file is executed on every boot (including wake-boot from deepsleep)   
#import esp                                                                  
#esp.osdebug(None)                                                           
#import webrepl                                                              
#webrepl.start()  
```

**Reply by robert-hh** [reply]
I uploaded you script to main.py on a ESP32C3 super mini, and it starts if I switch power at the USB hub port. It starts as well If I power the board by the GND and 5V Pin.

**Reply by bcux** [reply]
This is very strange:
 I tested 2 devices, same problem. Micropython 1.22.2, 1.22.0, 1.21.0 and 1.20.0, different USB cables.
I think something is wrong with the boards
![esp32-c3](https://github.com/micropython/micropython/assets/162479615/008dd811-30be-4700-a32a-d92b403f538b)

**Reply by robert-hh** [reply]
Your board looks identical to the boards I used for testing.

**Reply by bcux** [reply]
I found the cause of the problem and a solution:

I already suspected yesterday that it could be a hardware problem, both ESP32-C3 come from the same delivery and have similar MACs.
The fact that the program only failed to start during power-up also pointed to this. 
Since I also work with microcontrollers such as Microchip PIC, I know that missing or unsuitable capacitors are the cause of malfunctions.
I therefore first plugged a wired capacitor into the connections and - bingo - the problem was solved. Now a MLCC is soldered to 3.3V and GND - 100 or 220 nF - and everything is ok.
![c1](https://github.com/micropython/micropython/assets/162479615/d428f264-c33d-4739-87a2-618202250321)
![c2](https://github.com/micropython/micropython/assets/162479615/61e39a3e-3838-4c92-8d64-ddbe70526199)

**Reply by ricksorensen** [reply]
Hi - I have a learning question (not being a very skilled electrical circuit reader.)

I presume the added cap is to quiet the 3v line.  Since the REPL worked after completing boot but not the the loading of main.py on power up does this mean it took that long to stabilize power to chip/flash?  

I note the schematic indicates a 10uF cap between 3v and ground:
![image](https://github.com/micropython/micropython/assets/3868608/d65bfd2e-0669-443c-9f7b-3d1b2a3b6e96)
Is this missing?

Thanks for the information.

**Reply by robert-hh** [reply]
There are two caps on my board at the output of the regulator, and the above picture shows them too. And they are connected to 3V3 and GND. Just from the size I cannot tell the value, 10µF are possible. If 100-220nF works as well, it seems a different problem than just missing caps.

**Reply by bcux** [reply]
I think, one of the capacitors ist damaged or the boards are equipped with a wrong value.. Robert has the same board and main.py runs on his boards, but not on my boards (from the same delivery). When I removed the wired capacitor, the problem appeared again. I'm satisfied, when the boards do their job with the capacitor.

**Reply by davefes** [reply]
@robert-hh  > If 100-220nF works as well, it seems a different problem than just missing caps.

Had a look at a schematic (maybe one possible variant) and there are 3 bypass capacitors UC5, UC6 and UC7 on the VDDP3 pins.  And another 10uF on the 3V3 regulator output.  The 5V line must run all across the board to get to the regulator ... where the 10uF (PC2) is located I can't see.  

A parts placement picture and the track layout on all layers might be useful.  

Seeing as the RST pin is in the area where the added cap helps I wonder it might be the ENABLE bypass issue we had on the ESP32?

**Reply by robert-hh** [reply]
There is a larger capacitor next to the 3V regulator, which is connected to 3V3 and GND. I assume that this ought to be the 10µ part. The Multimeter says 17µF. But that may be wrong, since the part is soldered.
The board does not have Reset (EN) exposed at the solder pads. The one next to the RST label is GPIO 4.

**Reply by davefes** [reply]
OK on RST!  If the non-ground end on the RST button is closest to GPIO 4 it still might be "close" to the area in question.

Another, perhaps relevant question, where is system ground?  On the shell of the USB connector?

**Reply by robert-hh** [reply]
The USB shell is connected to GND, and the GND pin is next to it.

**Reply by davefes** [reply]
Do all the regulator bypass caps return to that point by a copper ground plane or by thin tracks back to the USB connector?  I am diving-into layout details that may not even be relevant here.

I would want to understand why the added cap fixed things as that might suggest a bigger problem or a better solution.  Adding caps may fix one problem but could introduce other problems. 

https://www.google.com/url?sa=t&rct=j&q=&esrc=s&source=web&cd=&ved=2ahUKEwig5saVrOOEAxV12DgGHcrZAs0QFnoECBsQAQ&url=https%3A%2F%2Fwww.compuphase.com%2Felectronics%2FParallelCapacitors_Antiresonance.pdf&usg=AOvVaw1XyxXbtGMVCb8Ebh8CiymC&opi=89978449

**Reply by robert-hh** [reply]
This is a cheap board. 2 layers, as it seems without a ground plane. And the habit of having a black solder stop lacquer makes it hard to follow traces. So I stop here.

**Comment by bcux** [comment]
Yesterday i received 2 ESP32-C3 from a friend and his devices were fully functional. I'm now sure that my parts have a manufacturing defect.
